phonon/update.py

```diff
@@ -109,25 +102,26 @@ def force_expiry(self):
         self.set_force_expiry()
         self.end_session()
 
-    def end_session(self, block=True):
+    def end_session(self):
         """
         Indicate to this update its session has ended on the local machine.
         The implementation of your cache, merge, and execute methods will be
         used to write to redis or your database backend as efficiently as
         possible.
 
         """
-        if not self.ref.dereference(self._execute, block=block):
+        if not self.ref.dereference(self._execute):
             if self.is_expired():
                 # If this update has expired but other active references
                 # to the resource still exist, we force this update to
                 # execute. We reset the time_modified_key and cached data
                 # to prevent any other processes from executing the same
                 # data.
                 self._execute()
-                with self.ref.lock(block=block):
-                    self.__process.client.delete(self.ref.resource_key)
-                    self.__process.client.set(self.ref.times_modified_key, 0)
+
+                with self.ref.lock():
```

Comment:
This method now raises `AlreadyLocked` when the lock is already acquired. This means the responsibility to catch that exception falls on the `LruCache`. For right now it will `raise` and fail. In the next pull request the `AsyncLruCache` should define a `on_failure` method to allow users to retry failed updates. Generally speaking, though, we shouldn't need to lock at all as we'll be defining some pretty granular methods for conflict free updates.